tc/sgx/trusted_worker_manager/enclave_wrapper/file_io_processor.cpp

```diff
@@ -0,0 +1,298 @@
+            result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+            fileStream.close();
+            return FAILED;
+        }
+
+       tcf::Log(TCF_LOG_ERROR, "File %s read has been completed\n", fileName.c_str());
+
+       // Copy the read file content to output buffer
+       outBuf = (uint8_t *)strncpy((char *)outBuf, buff, fileLen);
+       outBufSize = fileLen;
+       delete[] buff;
+       fileStream.close();
+
+    } catch (std::ifstream::failure e) {
+         tcf::Log(TCF_LOG_ERROR, "Caught exception while Reading file: %s\n", e.what());
+         resultStr = "FAILED TO READ: Caught Exception while reading file";
+         resultSize = resultStr.length();
+         result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+         return FAILED;
+    }
+
+    resultStr = "FILE READ SUCCESS";
+    resultSize = resultStr.length();
+    result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+
+    return SUCCESS;
+}
+
+// Writes file with the content of the buffer and updates status
+// SUCCESS or FAILED in result parameter
+uint32_t FileWrite(std::string fileName, uint8_t *result, size_t resultSize,
+    const uint8_t *inBuf, size_t inBufSize) {
+    string resultStr;
+    if (fileName.empty() || inBuf == nullptr) {
+        tcf::Log(TCF_LOG_ERROR, "Filename is empty\n");
+        resultStr = "FAILED TO WRITE: Filename is empty";
+        resultSize = resultStr.length();
+        result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+        return FAILED;
+    }
+
+    try {
+        ofstream fileStream (fileName);
+        if (!fileStream.is_open()) {
+            tcf::Log(TCF_LOG_ERROR, "Unable to open file %s\n", fileName.c_str());
+            resultStr = "FAILED TO WRITE: Unable to open file";
+            resultSize = resultStr.length();
+            result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+            return FAILED;
+        }
+
+        // write buffer data to file
+        fileStream.write((const char*)inBuf, inBufSize);
+        tcf::Log(TCF_LOG_INFO, "File %s WRITE has been completed\n", fileName.c_str());
+        fileStream.close();
+
+    } catch (std::ofstream::failure e) {
+        tcf::Log(TCF_LOG_ERROR, "Caught exception while Writing file: %s\n", e.what());
+        resultStr = "FAILED TO WRITE: Caught Exception while writing file";
+        resultSize = resultStr.length();
+        result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+        return FAILED;
+    }
+
+    resultStr = "FILE WRITE SUCCESS";
+    resultSize = resultStr.length();
+    result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+    return SUCCESS;
+}
+
+// Tells current position of the file pointer and stores it in buffer
+// updates status SUCCESS or FAILED in result parameter
+uint32_t FileTell(std::string fileName, uint8_t *result, size_t resultSize,
```

Comment:
What is the purpose of the `resultSize` parameter? It is unused. The incoming parameter value is always overwritten in the function and the incoming value is never used.
You should test that the string copied to `result` never exceeds the input value of `resultSize` (including the `NULL` character).
Also document in in the function comments that `resultSize` is the maximum size of the `result` buffer.

Reply:
I agree, we need to use resultSize as max threshold value of the string value that can be copied to result parameter. 
I will add this change along with documentation.